# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/ClientCanvasLayer.h
# Commit: 16c5dd2cfd60
# Full Hash: 16c5dd2cfd600a62c80ca17cfb35c0868e016056
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r ca8139d58d58 -r 16c5dd2cfd60 gfx/layers/client/ClientCanvasLayer.h
--- a/gfx/layers/client/ClientCanvasLayer.h	Fri Jun 12 21:53:14 2020 +0000
+++ b/gfx/layers/client/ClientCanvasLayer.h	Mon Jun 15 18:25:55 2020 +0000
@@ -30,7 +30,7 @@
     MOZ_COUNT_CTOR(ClientCanvasLayer);
   }
 
-  CanvasRenderer* CreateCanvasRendererInternal() override;
+  RefPtr<CanvasRenderer> CreateCanvasRendererInternal() override;
 
  protected:
   virtual ~ClientCanvasLayer();
@@ -59,7 +59,7 @@
   Layer* AsLayer() override { return this; }
   ShadowableLayer* AsShadowableLayer() override { return this; }
 
-  void Disconnect() override { mCanvasRenderer->Destroy(); }
+  void Disconnect() override { mCanvasRenderer->DisconnectClient(); }
 
   CompositableClient* GetCompositableClient() override {
     ClientCanvasRenderer* canvasRenderer =